src/main/java/com/dataloom/search/SearchApi.java

```diff
@@ -64,7 +66,22 @@
     SearchResult executeEntitySetDataQuery(
             @Path( ENTITY_SET_ID ) UUID entitySetId,
             @Body SearchTerm searchTerm );
-    
+
+    /**
+     * Executes a search over the data of a given entity set to find rows matching the specified property type values
+     * 
+     * @param entitySetId The id of the entity set the search will be executed within
+     * @param search A JSON object that contains three parameters: "start", which specifies the hit number to start
+     *            returning results on for paging, "maxHits", which specifies the maximum number of hits to return, and
+     *            "searchFields", which is a map from property type ids to search terms to match on those property
+     *            types.
+     * @return A search result object, containing the total number of hits for the given query, and the hits themselves
+     */
+    @POST( BASE + ADVANCED + ENTITY_SET_ID_PATH )
+    SearchResult executeAdvancedEntitySetDataQuery(
```

Comment:
potentially--my only concern with this is that even though they're very similar, there's no overlap between them. In the entity set metadata search it makes more sense to combine all the optional fields into a single object since a combination of the optional fields executes a combined search (like for matching entity types that also match on a keyword), and this doesn't really apply to the data search, and they also execute different types of queries. But yeah from a functionality perspective it makes sense to consolidate them. So I'm not sure what best practices are here/how to weight the importance of things